core/src/index.tsx

```diff
@@ -22,13 +23,62 @@ import routeInfoStore from './stores/route';
 import { emit } from './utils/event-hub';
 import browserHistory from './history';
 import { setConfig } from './config';
+import { initModuleFederationModule, useDynamicScript } from './utils/mf-helper';
 
-const holderDractDom = ReactDom; // if not use it, minified build file will cause infinite loop when use ReactDom.render. errMsg: Cannot set property 'getCurrentStack' of undefined
+// @ts-ignore if not use it, minified build file will cause infinite loop when use ReactDom.render. errMsg: Cannot set property 'getCurrentStack' of undefined
+const holderDractDom = ReactDom;
 
 setConfig('history', browserHistory);
 
+const enterpriseModule = [
+  {
+    name: 'fdp',
+    routerPrefix: '/fdp',
+  },
+  {
+    name: 'admin',
+    routerPrefix: '/sysAdmin',
+  },
+];
+
+const matchEnterpriseRoute = () => {
+  const target = enterpriseModule.find(({ routerPrefix }) => {
+    const routeRegex = new RegExp(`^/[^/]+${routerPrefix}`);
+    if (routeRegex.test(location.pathname)) {
+      return true;
+    }
+    return false;
+  });
+  return target;
+};
+
 const App = () => {
   const route = routeInfoStore.useStore((s) => s.parsed);
+  let location = useLocation();
```

Comment:
Suggest to use location from `browserHistory.listen` below

Reply:
I've tried this way, `browserHistory.listen` has one problem that it wont trigger callback for the first mount.
on other hand, we have set AutomaticVendorFederation for mf, So I guess import `react-use` in core is ok, it may share with shell